core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -309,22 +307,55 @@ public Envelope computeIteration(Envelope base,
 
         // 1) compute the potential junction parts (slowdown or speedup)
         var leftPart = computeLeftJunction(base, coreEnvelope, imposedBeginSpeed);
+
         var leftPartEndPos = leftPart != null ? leftPart.getEndPos() : base.getBeginPos();
         var coreEnvelopeWithLeft = computeEnvelopeWithLeftJunction(base, coreEnvelope, leftPart);
 
         var rightPart = computeRightJunction(base, coreEnvelopeWithLeft, imposedEndSpeed);
         var rightPartBeginPos = rightPart != null ? rightPart.getBeginPos() : base.getEndPos();
 
-        // if the junction parts touch or intersect, there is no core phase
-        if (rightPartBeginPos <= leftPartEndPos) {
+        if (EnvelopePhysics.areIntersecting(rightPart, leftPart)) {
+            // if the junction parts touch or intersect, there is no core phase
             return intersectLeftRightParts(leftPart, rightPart);
+        } else if (rightPartBeginPos < leftPartEndPos)
```

Comment:
can you move the if out of the else? it's pretty confusing, as all of the code below only runs when there is no intersection anyway